src/System.Management.Automation/FormatAndOutput/common/Utilities/Mshexpression.cs

```diff
@@ -193,30 +234,44 @@ internal List<MshExpression> ResolveNames(PSObject target, bool expand)
             return retVal;
         }
 
-        internal List<MshExpressionResult> GetValues(PSObject target)
+        /// <summary>
+        /// Gets the values of the object properties matched by this expression.
+        /// </summary>
+        /// <param name="target">The object to match against.</param>
+        public List<PSPropertyExpressionResult> GetValues(PSObject target)
         {
             return GetValues(target, true, true);
         }
 
-        internal List<MshExpressionResult> GetValues(PSObject target, bool expand, bool eatExceptions)
+        /// <summary>
+        /// Gets the values of the object properties matched by this expression.
+        /// </summary>
+        /// <param name="target">The object to match against.</param>
+        /// <param name="expand">If the matched properties are parameter sets, expand them.</param>
+        /// <param name="eatExceptions">If true, any exceptions that occur during the match process are ignored.</param>
+        public List<PSPropertyExpressionResult> GetValues(PSObject target, bool expand, bool eatExceptions)
         {
-            List<MshExpressionResult> retVal = new List<MshExpressionResult>();
+            List<PSPropertyExpressionResult> retVal = new List<PSPropertyExpressionResult>();
+
+            // If the object passed in is a hashtable, then turn it into a PSCustomObject so
+            // that property expressions can work on it.
+            target = IfHashtableWrapAsPSCustomObject(target);
```

Comment:
This could be a breaking change when it comes to the `ConstrainedLanguage` mode.
`@{prop = 3} | measure-object Count` works in `ConstrainedLanguage` mode today, but with this change, the conversion from `Hashtable` to `PSCustomObject` will throw exception.

Reply:
Hmm - true - good catch. It's pretty obscure and currently the only useful scenario for piping hashtables into `measure-object` (other than instance-counting). Still, we should probably get a committee review on this.